lib/url.c

```diff
@@ -2268,9 +2276,18 @@ CURLcode Curl_setopt(struct Curl_easy *data, CURLoption option,
      */
     data->set.buffer_size = va_arg(param, long);
 
-    if((data->set.buffer_size> (BUFSIZE -1)) ||
+    if((data->set.buffer_size > MAX_BUFSIZE) ||
        (data->set.buffer_size < 1))
-      data->set.buffer_size = 0; /* huge internal default */
+      data->set.buffer_size = MAX_BUFSIZE; /* huge internal default */
```

Comment:
Isn't it weird to get 512KB if you ask for < 1 ? Wouldn't the default size be more reasonable then?